src/sentry/services/hybrid_cloud/region.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from sentry.db.models import BaseManager
+from sentry.services.hybrid_cloud import ArgumentDict
+from sentry.services.hybrid_cloud.rpc import RpcServiceUnimplementedException
+from sentry.types.region import Region, get_region_by_name
+
+if TYPE_CHECKING:
+    from sentry.models import OrganizationMapping
+
+
+class RegionResolution(ABC):
+    """Interface for directing a service call to a remote region."""
+
+    @abstractmethod
+    def resolve(self, arguments: ArgumentDict) -> Region:
+        """Return the region determined by a service call's arguments."""
+        raise NotImplementedError
+
+    @staticmethod
+    def _resolve_from_mapping(mapping: OrganizationMapping) -> Region:
+        return get_region_by_name(mapping.region_name)
+
+    @property
+    def organization_mapping_manager(self) -> BaseManager[OrganizationMapping]:
+        from sentry.models import OrganizationMapping
+
+        # Convenience method to avoid repeating the local import
+        return OrganizationMapping.objects
+
+
+@dataclass(frozen=True)
+class ByOrganizationObject(RegionResolution):
+    """Resolve from a parameter representing an organization object."""
```

Comment:
Following up on [this comment](https://github.com/getsentry/sentry/pull/45448#discussion_r1128536536) from @markstory, now that this is done, the case where this object finds the right parameter by inference didn't come up as often I was expecting. The inference code plainly isn't worth it, so I got rid of it. (https://github.com/getsentry/sentry/pull/46638/commits/6a1b698c1d2fbaa0894cabe03d9343bb37bd798c)